### PR TITLE
crimson/osd: cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

File: src/crimson/admin/admin_socket.cc [hunk 1]
```diff
@@ -225,30 +225,29 @@ seastar::future<> AdminSocket::start(const std::string& path)
...
         });
+      }).handle_exception([this](auto ep) {
+        if (!stop_gate.is_closed()) {
+          logger().error("AdminSocket: terminated: {}", ep);
+        }
       });
-    }).finally([path] {
-      return seastar::remove_file(path);
     });
+  }).handle_exception_type([](const seastar::gate_closed_exception&) {
```
Comment (liu-chunmei) [inline src/crimson/admin/admin_socket.cc:247]: what should it do for the exception?

Comment (tchaikov) [reply src/crimson/admin/admin_socket.cc:247]: @liu-chunmei nothing, seastar uses it for changing the control flow. In this case, it breaks the loop if gate is closed. So we just need to catch and ignore the exception out of the loop.

**Review by liu-chunmei — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- s/error/info/ when printing "`crimson shutdown complete`".

**Pull request merged**